lib/fw.c

```diff
@@ -393,25 +408,36 @@ struct fw_image_header {
 int switchtec_fw_file_info(int fd, struct switchtec_fw_image_info *info)
 {
 	int ret;
-	struct fw_image_header hdr = {};
+	struct fw_image_header hdr;
+	unsigned char vend[5] = {"MSCC"};
+	unsigned char count;
+	unsigned char ver_num;
+	int i;
+
+	if (info == NULL)
+		return 0;
 
-	ret = read(fd, &hdr, sizeof(hdr));
 	lseek(fd, 0, SEEK_SET);
+	ret = read(fd, &hdr, sizeof(hdr));
 
 	if (ret != sizeof(hdr))
 		goto invalid_file;
 
-	if (strcmp(hdr.magic, "PMC") != 0)
-		goto invalid_file;
-
-	if (info == NULL)
-		return 0;
+	count = 0;
+	for (i = 0; i < 4; i++) {
+		if(hdr.vendor_id[i] == vend[i])
+			count ++;
+	}
 
-	info->type = hdr.type;
-	info->crc = le32toh(hdr.image_crc);
-	version_to_string(hdr.version, info->version, sizeof(info->version));
-	info->image_addr = le32toh(hdr.load_addr);
-	info->image_len = le32toh(hdr.image_len);
+	if (count == 4) {
+		info->type = (enum switchtec_fw_image_type)switch_convertu8tou32(hdr.part_type, 4);
+		info->crc = switch_convertu8tou32(hdr.img_crc, 4);
```

Comment:
why not keep original code for these assignment?

Reply:
saratoga define the fields by 'uint32_t', trident define them by 'unsigned char part_type[4]' mode, in order to get the correct byte endian mode, convert them to uint32_t by function.